src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs

```diff
@@ -319,29 +319,38 @@
           (nil? entity-types) [:div {:style {:textAlign "center"}} [comps/Spinner {:text "Retrieving entity types..."}]]
           :else
           (let [attribute-keys (->> entity-list (map #(-> (% "attributes") keys set)) (apply union))
-                attr-col-width (->> attribute-keys count (/ 1000) int (min 400) (max 100))]
+                attr-col-width (->> attribute-keys count (/ 1000) int (min 400) (max 100))
+                entity-column [{:header "Entity Name" :starting-width 200
+                                :as-text #(% "name") :sort-by :text
+                                :content-renderer (or (:entity-name-renderer props)
+                                                      (fn [entity] (entity "name")))}]
+                attr-columns (map (fn [k] {:header k :starting-width attr-col-width :sort-by :text
+                                           :as-text
+                                           (fn [attr-value]
+                                             (cond
+                                               (is-single-ref? attr-value) (attr-value "entityName")
+                                               (is-ref-list? attr-value) (map #(% "entityName") attr-value)
+                                               :else (str attr-value)))
+                                           :content-renderer
+                                           (fn [attr-value]
+                                             (cond
+                                               (is-single-ref? attr-value) (attr-value "entityName")
+                                               (is-ref-list? attr-value) (join ", " (map #(% "entityName") attr-value))
+                                               :else ((:attribute-renderer props) attr-value)))})
+                                  attribute-keys)
+                make-pair (fn [s]
+                            [{:header (str "# " s) :starting-width 110}
+                             #(count (get-in % ["attributes" s]))])
+                [count-column count-data] (cond (= selected-entity-type "sample_set") (make-pair "samples")
+                                                (= selected-entity-type "pair_set") (make-pair "pairs")
+                                                (= selected-entity-type "participant_set") (make-pair "participants"))
+                columns (if count-column
+                          (concat entity-column [count-column] attr-columns)
```

Comment:
Since attr-colums is a list, if you made entity-column a map (instead of a vector of a single item), you could do 
```clojure
(conj attr-colums count-column entity-column)
```
since lists conjoin on the head (vs. vectors which conjoin on the end).